base/commands/list/list.go

```diff
@@ -0,0 +1,74 @@
+//go:build base || list
+
+package list
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hazelcast/hazelcast-go-client"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	"github.com/hazelcast/hazelcast-commandline-client/clc/paths"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+)
+
+const (
+	listFlagName     = "name"
+	listFlagShowType = "show-type"
+	listPropertyName = "list"
+)
+
+type ListCommand struct {
+}
+
+func (mc *ListCommand) Init(cc plug.InitContext) error {
+	cc.SetCommandGroup(clc.GroupDDSID)
+	cc.AddStringFlag(listFlagName, "n", defaultListName, false, "list name")
+	cc.AddBoolFlag(listFlagShowType, "", false, false, "add the type names to the output")
+	if !cc.Interactive() {
+		cc.AddStringFlag(clc.PropertySchemaDir, "", paths.Schemas(), false, "set the schema directory")
+	}
+	cc.SetTopLevel(true)
+	cc.SetCommandUsage("list [command] [flags]")
+	help := "List operations"
+	cc.SetCommandHelp(help, help)
+	return nil
+}
+
+func (mc *ListCommand) Exec(context.Context, plug.ExecContext) error {
+	return nil
+}
+
+func (mc *ListCommand) Augment(ec plug.ExecContext, props *plug.Properties) error {
+	ctx := context.TODO()
```

Comment:
Is this line necessary?

Reply:
Do you mean do we need context? There are functions calls that require a context in that function. Could you elaborate if I understood wrong?

Reply:
I thought the `ctx` was a leftover.
We'll need to add context parameters to Set/GetBlocking calls.